Documentation/ReviewingGuidelines.txt

```diff
@@ -0,0 +1,116 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+If you are looking for a patch series in need of review, start by checking
+latest "What's cooking in git.git" email (example: [1]). Topics tagged with
+"Needs review" and those in the "[New Topics]" section are typically those that
+would benefit the most from additional review.
+
+If you've already contributed to Git, you may also be CC'd in another
+contributor's patch series. These are usually topics where the author feels that
+your attention is warranted; this may be due to prior contributions,
+demonstrated expertise, and/or interest in related topics. There is no
+requirement to review these series, but you may find them easier to review as a
+result of your preexisting background knowledge on the topic.
+
+Reviewing patches
+~~~~~~~~~~~~~~~~~
+While every contributor takes their own approach to reviewing patches, here are
+some general pieces of advice to make your reviews to be as clear and helpful as
+possible.
+
+- Provide your review comments per-patch in a plaintext "Reply-All" email to the
+  relevant patch. Comments should be made inline, immediately below the relevant
+  section(s).
+
+- Remember to review the content of commit messages for correctness and clarity,
+  in addition to the code change in the patch's diff. The commit message of a
+  patch should accurately and fully explain the code change being made in the
+  diff.
+
+- If a patch is long, you can delete irrelevant chunks of the patch from your
+  reply. Make sure you leave enough to provide the appropriate context for your
+  comments!
+
+- When pointing out an issue, try to include suggestions for how the author
+  could fix it. This not only helps the author to understand and fix the issue,
+  it also deepens and improves your understanding of the topic.
+
+- Reviews do not need to exclusively point out problems. Feel free to "think out
+  loud" in your review: describe how you read & understood a complex section of
+  a patch, ask a question about something that confused you, point out something
+  you found particularly well-written, etc.
+
+- When providing a recommendation, be as clear as possible about whether you
+  consider it "blocking" (the code would be broken or otherwise made worse if an
+  issue isn't fixed) or "non-blocking" (the patch could be made better by taking
+  the recommendation, but acceptance of the series does not require it).
+  Non-blocking recommendations can be particularly ambiguous when they are
+  related to - but outside the scope of - a series ("nice-to-have"s), or when
+  they represent only stylistic differences between the author and reviewer.
+
+- If you cannot complete a full review of a series all at once, consider letting
+  the author know (on- or off-list) if/when you plan to review the rest of the
+  series.
+
+- If you read and review a series but find nothing that warrants inline
+  commentary, reply to the series' cover letter to indicate that you've reviewed
+  the changes.
```

Comment:
I would like to suggest to add a few more bullet points:

- In many cases, the patches cannot be reviewed properly merely by looking at the diffs. For example, sometimes the default of three context lines in each diff hunk are insufficient to fully understand the intention of the patch. In such cases, applying those patches to a worktree and analyzing the full code (e.g. analyzing related parts of the code that are not shown in the diff, looking at the history of the code touched by the diff, building and running automated and manual tests) will help you provide much more helpful reviews.
- Often patches touch on areas that have been discussed before on the Git mailing list. It is helpful in such cases to find the link to the Git mailing list archive and most importantly to summarize the outcome(s) of the previous conversation.
- Reviews are most helpful when they are constructive, i.e. when they are designed to make the patch(es) better. For example, while it is rarely helpful to offer personal opinions, contributions often benefit from reviewers' insights about avoidable coding patterns and how to adjust the code according to best-known practices. Especially new contributors benefit from reviews that help adjust the code to Git's customs (preferably with an explanation why those customs exist).
- Be kind and graceful in your reviews. Keep in mind that the recipient of your review is a human being, not a machine. Ask only as much effort of the contributors as you are prepared to give. Lead by example, e.g. show the very thoroughness and diligence in your review that you would like to see in the contribution. If you ask the contributor to "do their homework", you can't just say "I haven't fully read the patch series yet but [...]" in the same mail.

Reply:
More ideas to consider:

* To review patches locally in your worktree, you can apply the patches to your repository or wait for the branch to be tracked by the maintainer and check out that version.

* Some patches that perform significant refactoring can result in hard-to-read patches that become easier to read as commit diffs using `--color-moved` or `--ignore-space-change`.

* It is important to consider whether the changed code is covered by existing tests, or if a new test is required to exercise that behavior. With a local checkout of the patches, you can mutate the lines to be obviously wrong and see if that triggers a test failure. If you see a lack of testing, then recommend methods to test that behavior, including by writing a test yourself.